src/redux/utils/api.js

```diff
@@ -37,19 +43,23 @@ function request (method = 'GET', path: string, payload?: FormData|Object): Func
       }
     }
 
-    return fetch(API_BASE + path, options)
-      .then((res) => res.json().then((json) => ({ res, json })))
-      .then(({ res, json }) => {
-        if (!res.ok) {
-          if (token && res.status === 401) {
-            dispatch(discardToken())
-          }
+    return fetch(API_BASE + path, options).then((response) => {
+      if (response.ok) {
+        return containsJson(response) ? response.json() : response
+      } else {
+        if (token && response.status === 401) {
+          dispatch(discardToken())
+        }
 
-          return Promise.reject(json)
+        if (containsJson(response)) {
+          return response.json().then((json) => Promise.reject(json))
         }
 
-        return json
-      })
+        var error = new Error(response.statusText)
+        error.response = response
+        throw error
+      }
+    })
```

Comment:
You really want to forbid any other type or response than json? I don't think it would be "bad" if the user receive a non-json response. It's up to the client to handle it (or not handle it...)